### PR TITLE
feat: add module hook to add make commands

### PR DESCRIPTION
**Pull request by tmerkel**
## What this PR does / why we need it
Adds a module hook to allow other modules to add Make commands to the root Makefile.

Allows `stencil-graphql` to add a command for running the rover-cli for dev work.

## Jira ID

[XX-XX]

### CODE REVIEW

File: docs/module-hooks.md [hunk 1]
```diff
@@ -175,3 +175,20 @@ Extra [devspace profiles](https://www.devspace.sh/docs/5.x/configuration/profile
...
+
+### `Makefile.commands`
+
+**Type**: `string`
+
+**File**: `Makefile`
+
+Extra commands to add to the root Makefile
+
+```tpl
```
Comment (artemchernyak-outreach) [inline docs/module-hooks.md:187]: This example should include an actual sample of how you would apply this. This should include the call to `stencil.AddToModuleHook`. 

Ideally, this example should come from your test case of running this inside a stencil module. You can always use `replacement` inside `service.yaml` to test this locally.

Comment (tmerkel) [reply docs/module-hooks.md:187]: ...I failed to capture the surrounding context :(. Thanks.  Let me try that again.

Comment (tmerkel) [reply docs/module-hooks.md:187]: Better?

Comment (artemchernyak-outreach) [reply docs/module-hooks.md:200]: Perfect! Thanks for expanding this.

**Review by artemchernyak-outreach — APPROVED** [review]

**Pull request merged**